### PR TITLE
harden `ClusterLogSpec`

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

The `JoinAsync` method wasn't written correctly - need to force `JoinAsync` to run to completion for assertion to make any kind of sense.

### CODE REVIEW

File: src/core/Akka.Cluster.Tests/ClusterLogSpec.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ protected async Task JoinAsync(string expected)
             {
                 await EventFilter
                     .Info(contains: expected)
-                    .ExpectOneAsync(async () => _cluster.Join(_selfAddress));
+                    .ExpectOneAsync(async () => await _cluster.JoinAsync(_selfAddress).ShouldCompleteWithin(Remaining));
```
Comment (Arkatufus) [inline src/core/Akka.Cluster.Tests/ClusterLogSpec.cs:71]: I have little trust that the `JoinAsync` method is actually correct. I think we should use a more manual way of asserting this, like the one you've implemented in the `DownAsync` method below.

**Review by Arkatufus — CHANGES_REQUESTED** [review]
Just a slight change needed

File: src/core/Akka.Cluster.Tests/ClusterLogSpec.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aaronontheweb) [reply src/core/Akka.Cluster.Tests/ClusterLogSpec.cs:71]: Will do

[the remark this answers is not in the document]
Comment (Aaronontheweb) [reply src/core/Akka.Cluster.Tests/ClusterLogSpec.cs]: Done

**Review by Arkatufus — APPROVED** [review]
LGTM

**Pull request merged**